### ISSUE TITLE
Unable to deploy VM using Template and custom disk offering

### ISSUE THREAD

**Issue by sbrueseke** [apache/cloudstack#7115]
##### ISSUE TYPE

 * Bug Report

##### COMPONENT NAME

~~~
VM
~~~

##### CLOUDSTACK VERSION

~~~
4.17.2.0
~~~

##### CONFIGURATION

hypervisor: xcp-ng, advanced networking

##### SUMMARY

Unable to deploy VM using Template and custom disk offering, fails with error "VM Creation failed. Volume size: -1GB is out of allowed range. Max: 1024 Min:1". This issue is similar to this issue which for ISO: https://github.com/apache/cloudstack/issues/6440 but not the same.

##### STEPS TO REPRODUCE

1) Add a new Disk Offering with customized = true
`/client/api/?isMirrored=false&name=Test+Custom+Disk+Offering&displaytext=Test+Custom+Disk+Offering&storageType=shared&cacheMode=none&provisioningType=thin&customized=true&disksizestrictness=false&command=createDiskOffering&response=json
`
2) Create a new Compute Offering with diskofferingstrictness=true (please be aware that in 4.17.2.0 there is a bug and you cannot do this via UI. There is a button for this, but it is broken. This has already being fixed https://github.com/apache/cloudstack/pull/7072) You need to add the disk offering uuid in this api call!
`/client/api/?issystem=false&name=Test+Compute+Offering&displaytext=Test+Compute+Offering&customized=false&offerha=false&limitcpuuse=false&dynamicscalingenabled=false&diskofferingid=<uuid>&cpunumber=1&cpuspeed=500&memory=512&command=createServiceOffering&diskofferingstrictness=true&response=json`

3) Go to the instance wizard in the UI and choose the just created Compute offering. Do not click anything else and go straight to “Launch instance”.

##### EXPECTED RESULTS

UI needs to pass parameter rootdisksize with a default value in header of api call deployVirtualMachine even if "Override root disk size" in Template/ISO section is disabled. I do not know where do get this default value from.

##### ACTUAL RESULTS

UI is not passing parameter rootdisksize via header when using api call deployVirtualMachine and "Override root disk size" in Template/ISO section is disabled. When enabling "Override root disk size" in Template/ISO section a default value of 10 is inserted into the wizard and it works.

**Comment by sbrueseke** [comment]
A question in my opinion is if a custom disk offering should have a default or minimum size value attached to it. I am nit sure where the value of 10 is coming from, but I suppose from the used template parameters.

**Comment by weizhouapache** [comment]
@sbrueseke 
I can reproduce the issue on UI.

The default value of root disk size comes from the virtual size (not physical size) of vm template. see example below
```
(localcloud) 🐱 > list templates templatefilter=all filter=size,physicalsize
{
  "count": 4,
  "template": [
    {
      "physicalsize": 461307904,
      "size": 5242880000
    },
    {
      "physicalsize": 43646976,
      "size": 52428800
    },
    {
      "physicalsize": 453181440,
      "size": 4194304000
    },
    {
      "physicalsize": 1769537536,
      "size": 8589934592
    }
  ]
}
```

**Comment by sbrueseke** [comment]
@weizhouapache does it make sense to add the default template size to the api call if another value is not provided?

**Comment by weizhouapache** [comment]
@sbrueseke yes, it makes sense.
I think it should be obliged to set the root disk size when disk offering is dynamic and strictness is true.

a UI bug

**Comment by sbrueseke** [comment]
@weizhouapache yes, it should be, but which is the default value to put into the wizard? The size of the template?

**Comment by weizhouapache** [comment]
> @weizhouapache yes, it should be, but which is the default value to put into the wizard? The size of the template?

yes @sbrueseke 
If vm is created from ISO, the default might be 1

**Comment by weizhouapache** [comment]
fyi, the workaround is as below
![image](https://github.com/apache/cloudstack/assets/57355700/3cccaafa-3c19-4ada-bb52-aeba62749339)
check "Override root disk size" , then it works

**Comment by weizhouapache** [comment]
marked this as a minor issue

**Comment by weizhouapache** [comment]
fixed by #7913
